### PR TITLE
Update docs for newly added locking functions

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

Update based on https://github.com/pingcap/tidb/pull/44495 / https://github.com/pingcap/tidb/issues/44493

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: functions-and-operators/locking-functions.md [hunk 1]
```diff
@@ -12,16 +12,15 @@ TiDB supports most of the user-level [locking functions](https://dev.mysql.com/d
...
 |:---------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------|
 | [`GET_LOCK(lockName, timeout)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_get-lock)    | Acquires an advisory lock. The `lockName` parameter must be NO longer than 64 characters. Waits maximum `timeout` seconds before timing out and returns a failure.         |
-| [`RELEASE_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_release-lock)     | Releases a previously acquired lock. The `lockName` parameter must be NO longer than 64 characters. |
+| [`IS_FREE_LOCK()`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-free-lock) | Tests if a lock is free. |
+| [`IS_USED_LOCK()`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-used-lock) | Tests if a lock is used. Returns connection id if true. |
```
Comment (Oreoxmt) [inline functions-and-operators/locking-functions.md:16]:
```suggestion
| [`IS_USED_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-used-lock) | Checks if a lock is in use. If true, it returns the corresponding  connection ID. |
```

File: functions-and-operators/locking-functions.md [hunk 2]
```diff
@@ -12,16 +12,15 @@ TiDB supports most of the user-level [locking functions](https://dev.mysql.com/d
 | Name                                                                                                                 | Description                                                           |
 |:---------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------|
 | [`GET_LOCK(lockName, timeout)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_get-lock)    | Acquires an advisory lock. The `lockName` parameter must be NO longer than 64 characters. Waits maximum `timeout` seconds before timing out and returns a failure.         |
-| [`RELEASE_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_release-lock)     | Releases a previously acquired lock. The `lockName` parameter must be NO longer than 64 characters. |
+| [`IS_FREE_LOCK()`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-free-lock) | Tests if a lock is free. |
```
Comment (Oreoxmt) [inline functions-and-operators/locking-functions.md:15]:
```suggestion
| [`IS_FREE_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-free-lock) | Checks if a lock is free. |
```

**Comment by Oreoxmt** [comment]
@dveeden Could you please invite a tech reviewer? Thanks!

**Review by Oreoxmt — APPROVED** [review]

File: functions-and-operators/locking-functions.md [hunk 3]
```diff
@@ -12,16 +12,13 @@ TiDB supports most of the user-level [locking functions](https://dev.mysql.com/d
...
 |:---------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------|
 | [`GET_LOCK(lockName, timeout)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_get-lock)    | Acquires an advisory lock. The `lockName` parameter must be NO longer than 64 characters. Waits maximum `timeout` seconds before timing out and returns a failure.         |
-| [`RELEASE_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_release-lock)     | Releases a previously acquired lock. The `lockName` parameter must be NO longer than 64 characters. |
+| [`IS_FREE_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-free-lock) | Checks if a lock is free. |
+| [`IS_USED_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-used-lock) | Checks if a lock is in use. If true, it returns the corresponding  connection ID. |
```
Comment (Oreoxmt) [inline functions-and-operators/locking-functions.md:16]:
```suggestion
| [`IS_USED_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-used-lock) | Checks if a lock is in use. If true, it returns the corresponding connection ID. |
```

**Review by hfxsd — APPROVED** [review]

File: functions-and-operators/locking-functions.md [hunk 4]
```diff
@@ -12,16 +12,13 @@ TiDB supports most of the user-level [locking functions](https://dev.mysql.com/d
...
 |:---------------------------------------------------------------------------------------------------------------------|:----------------------------------------------------------------------|
 | [`GET_LOCK(lockName, timeout)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_get-lock)    | Acquires an advisory lock. The `lockName` parameter must be NO longer than 64 characters. Waits maximum `timeout` seconds before timing out and returns a failure.         |
-| [`RELEASE_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_release-lock)     | Releases a previously acquired lock. The `lockName` parameter must be NO longer than 64 characters. |
+| [`IS_FREE_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-free-lock) | Checks if a lock is free. |
+| [`IS_USED_LOCK(lockName)`](https://dev.mysql.com/doc/refman/5.7/en/locking-functions.html#function_is-used-lock) | Checks if a lock is in use. If true, it returns the corresponding connection ID. |
```
Comment (mjonss) [inline functions-and-operators/locking-functions.md:16]: Does it always return corresponding connection ID, or only if the same session (or TiDB node?) has the lock?

Comment (dveeden) [reply functions-and-operators/locking-functions.md:16]: c4978a35fde1949528e2add84b86d329b3436980

**Review by mjonss — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

File: functions-and-operators/locking-functions.md [hunk 5]
```diff
@@ -12,16 +12,14 @@ TiDB supports most of the user-level [locking functions](https://dev.mysql.com/d
...
 
 * The minimum timeout permitted by TiDB is 1 second, and the maximum timeout is 1 hour (3600 seconds). This differs from MySQL, where both 0 second and unlimited timeouts (`timeout=-1`) are permitted. TiDB will automatically convert out-of-range values to the nearest permitted value and convert `timeout=-1` to 3600 seconds.
 * TiDB does not automatically detect deadlocks caused by user-level locks. Deadlocked sessions will timeout after a maximum of 1 hour, but can also be manually resolved by using `KILL` on one of the affected sessions. You can also prevent deadlocks by always acquiring user-level locks in the same order.
 * Locks take effect on all TiDB servers in the cluster. This differs from MySQL Cluster and Group Replication where locks are local to a single server.
-
-## Unsupported functions
-
-* `IS_FREE_LOCK()`
-* `IS_USED_LOCK()`
+* `IS_USED_LOCK()` returns `1` if it is called from another session and is unable to return process ID of the process that is holding the lock.
```
Comment (hfxsd) [inline functions-and-operators/locking-functions.md:25]:
```suggestion
* `IS_USED_LOCK()` returns `1` if it is called from another session and is unable to return the ID of the process that is holding the lock.
```

**Pull request merged**